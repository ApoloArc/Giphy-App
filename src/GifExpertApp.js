import React, {useState} from 'react'
import GifGrid from './Components/GifGrid';
import { SearchBar } from './Components/SearchBar';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch Man']);

    // const sumarCategoria = () => {
    //     setCategories(categories =>[...categories, 'El Chavo del 8'])
    // }

    return (  
        <>
          <h1>Gif Expert App</h1>
          <SearchBar setCategories={setCategories}/>
          <ol>
            {
                categories.map(categoria => <GifGrid key={categoria} category ={categoria}/>)
            }
          </ol>
        </>
    )
}
