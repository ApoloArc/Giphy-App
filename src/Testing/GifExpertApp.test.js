import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Haciendo pruebas en el Expert App', () => {
    
    const wrapper = shallow(<GifExpertApp/>)

    test('Tomar el snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    })
    
    test('Debe de pasar las categories que se le manden', () => {
        
        const categories = ['Dragon Ball', 'El Chavo del 8'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    

})
