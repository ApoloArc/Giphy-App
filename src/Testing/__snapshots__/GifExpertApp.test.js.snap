// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas en el Expert App Debe de pasar las categories que se le manden 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <SearchBar
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="El Chavo del 8"
      key="El Chavo del 8"
    />
  </ol>
</Fragment>
`;

exports[`Haciendo pruebas en el Expert App Tomar el snapshot 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <SearchBar
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
